tests/conftest.py

```diff
@@ -63,6 +70,27 @@ def log_has_re(line, logs):
                   False)
 
 
+def num_log_contains(line, logs, count=False):
+    # caplog mocker returns log as a tuple: ('freqtrade.something', logging.WARNING, 'spamfoobar')
+    # and we want to check how many times line ('foo', for example) is contained in the tuples.
+    return sum(line in logger_name or line in message
+               for logger_name, level, message in logs.record_tuples)
+
+
+def num_log_has(line, logs, count=False):
+    # caplog mocker returns log as a tuple: ('freqtrade.something', logging.WARNING, 'foobar')
+    # and we want to check how many times line is presente in the tuples.
+    return sum(line == logger_name or line == message
+               for logger_name, level, message in logs.record_tuples)
+
+
+def num_log_has_re(line, logs, count=False):
+    # caplog mocker returns log as a tuple: ('freqtrade.something', logging.WARNING, 'foobar')
+    # and we want to check how many times line matches in the tuples.
+    return sum(re.match(line, logger_name) or re.match(line, message)
+               for logger_name, level, message in logs.record_tuples)
```

Comment:
this method is not used, which is hiding the fact that it's not working.
it'll fail with `unsupported operand type(s) for +: 'int' and 'NoneType'` if provided with a match which does not match (i suspect it'll also fail in case of matches - as sum expects a "int-like" value (it works with bools as they default to 1/0 - but not for "re.match" - which is not a boolean flag)

I've added some fixes to this (and reduced the amount of methods).
while it's not wrong to have them - i think it'll be confusing as to "when to use which" - and the performance impact of using `re.match` over `in` is neglectible in case of tests.